eos/utils/kmatrix-impl.hh

```diff
@@ -0,0 +1,213 @@
+                                               std::initializer_list<std::shared_ptr<KMatrix::Resonance>> resonances,
+                                               std::vector<std::vector<Parameter>> bkgcst,
+                                               const std::string & prefix) :
+    _channels(channels),
+    _resonances(resonances),
+    _bkgcst(bkgcst),
+    _prefix(prefix),
+    _Khat(gsl_matrix_complex_calloc(nchannels_, nchannels_)),
+    _That(gsl_matrix_complex_calloc(nchannels_, nchannels_)),
+    _tmp_1(gsl_matrix_complex_calloc(nchannels_, nchannels_)),
+    _tmp_2(gsl_matrix_complex_calloc(nchannels_, nchannels_)),
+    _perm(gsl_permutation_calloc(nchannels_))
+    {
+        // Perform size checks
+        if (channels.size() != nchannels_)
+            throw InternalError("The size of the channels array does not match nchannels_.");
+        if (bkgcst.size() != nchannels_)
+            throw InternalError("The size of the array of background constants does not match nchannels_.");
+        if (bkgcst.size() != 0 and bkgcst[0].size() != nchannels_)
+            throw InternalError("The array of background constants is not square.");
+        if (resonances.size() != nresonances_)
+            throw InternalError("The size of the resonances array does not match nresonances_.");
+
+        // Perform pointer checks
+        if (_Khat == nullptr)
+            throw std::bad_alloc();
+        if (_That == nullptr)
+            throw std::bad_alloc();
+        if (_tmp_1 == nullptr)
+            throw std::bad_alloc();
+        if (_tmp_2 == nullptr)
+            throw std::bad_alloc();
+        if (_perm == nullptr)
+            throw std::bad_alloc();
+    }
+
+    template <unsigned nchannels_, unsigned nresonances_>
+    KMatrix<nchannels_, nresonances_>::~KMatrix()
+    {
+        if (_perm)
+            gsl_permutation_free(_perm);
+        _perm = nullptr;
+
+        if (_tmp_1)
+            gsl_matrix_complex_free(_tmp_1);
+        _tmp_1 = nullptr;
+
+        if (_tmp_2)
+            gsl_matrix_complex_free(_tmp_2);
+        _tmp_2 = nullptr;
+
+        if (_That)
+            gsl_matrix_complex_free(_That);
+        _That = nullptr;
+
+        if (_Khat)
+            gsl_matrix_complex_free(_Khat);
+        _Khat = nullptr;
+    }
+
+
+    //Return row n° rowindex of the T matrix defined as T = (1-i*rho*K)^(-1)*K
+    template <unsigned nchannels_, unsigned nresonances_>
+    std::array<complex<double>, nchannels_>
+    KMatrix<nchannels_, nresonances_>::tmatrix_row(unsigned rowindex, double s) const
+    {
+
+        std::array<complex<double>, nchannels_> tmatrixrow;
+        auto channels = this->_channels;
+        auto resonances = this->_resonances;
+        auto bkgcst = this->_bkgcst;
+
+
+        ///////////////////
+        //1. Fill tmp1 = rho matrix
+        ///////////////////
+        gsl_matrix_complex_set_zero(_tmp_1);
+
+        for (size_t i = 0 ; i < nchannels_ ; ++i)
+            {
```

Comment:
Please align scope braces with their statement, i.e., here: the braces should be identically indented as the ```for```